api/task.go

```diff
@@ -152,6 +160,15 @@ type TaskDatabaseDataUpdatePayload struct {
 	BinlogFileEnd   string `json:"binlogFileEnd,omitempty"`
 	BinlogPosStart  int64  `json:"binlogPosStart,omitempty"`
 	BinlogPosEnd    int64  `json:"binlogPosEnd,omitempty"`
+	// RollbackTaskState is a state machine
+	// 0. The initial state is "".
+	// 1. When the rollback generation starts, we set it to "RUNNING".
+	// 2. When the rollback generation succeeds, we set it to "SUCCESS".
+	// 4. When the rollback generation fails, we set it to "FAIL", and set the err to RollbackError.
+	RollbackTaskState RollbackTaskState `json:"rollbackTaskState,omitempty"`
+	RollbackError     string            `json:"rollbackError,omitempty"`
+	// RollbackStatement is the generated rollback SQL statement for the DML task.
+	RollbackStatement string `json:"rollbackStatement,omitempty"`
```

Comment:
Maybe okay for now, but worth considering generating a gigantic SQL from deleting all table data.

Reply:
Add a TODO for now. We should generate on-the-fly if the generator finds that the generated data is more than a threshold of lines.

Reply:
Per offline discussion, we only allow generating rollback SQL < 8MB.